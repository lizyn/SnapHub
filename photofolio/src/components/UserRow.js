import React, { Component } from 'react';


function UserRow(props) {
  const {profilePicUrl, name, ring, showFollow} = props;
  // const profilePicUrl = defaultAvatar;
  // const username = props.name;
  return (
    <div className='user-row'>
      <div className='container'>
        {ring &&
          <img className='profile-pic profile-pic-ring' src={profilePicUrl} alt={name}/>
        }
        {!ring && 
          <img className='profile-pic' src={profilePicUrl} alt={name}/>
        }
        <div className='username'>{name}</div>
        {showFollow &&
          <button className='white-button right' onClick={follow}>
          Follow
          </button>
        }
      </div>
    </div>
  );
}

function follow(username) {

}

export default UserRow;