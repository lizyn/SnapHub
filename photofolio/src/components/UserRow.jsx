import React, { useState } from 'react';
import Button from '@mui/material/Button';
import PropTypes from 'prop-types';
import { rootUrl } from './Config';
import axios from '../api/axios';

function UserRow(props) {
  UserRow.propTypes = {
    avatar: PropTypes.string,
    name: PropTypes.string.isRequired,
    ring: PropTypes.bool,
    showFollow: PropTypes.bool,
    userId: PropTypes.number.isRequired
  };

  UserRow.defaultProps = {
    avatar:
      'https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1185.jpg',
    ring: false,
    showFollow: false
  };
  const { avatar, name, ring, showFollow, userId } = props;
  const [followBtn, setFollowBtn] = useState('outlined');
  const [followed, setFollowed] = useState(false);
  const [followId, setFollowId] = useState();
  const currentUserId = 1;
  const toggleFollow = async (userIdToFollow, isFollow) => {
    const params = {
      userId: userIdToFollow,
      followerId: currentUserId
    };
    try {
      if (isFollow) {
        const response = await axios.post(`${rootUrl}/follows`, params);
        setFollowId(response.data.id);
        console.log(response);
      } else if (followId) {
        const response = await axios.delete(`${rootUrl}/follows/${followId}`);
        console.log(response);
      }
    } catch (err) {
      console.error(err);
    }
  };
  const handleFollowBtnClick = () => {
    if (followed) {
      toggleFollow(userId, false);
      setFollowBtn('outlined');
    } else {
      toggleFollow(userId, true);
      setFollowBtn('contained');
    }
    setFollowed((o) => !o);
  };

  return (
    <div className="user-row">
      <div className="container">
        {ring && (
          <img
            className="profile-pic profile-pic-ring"
            src={avatar}
            alt={name}
          />
        )}
        {!ring && <img className="profile-pic" src={avatar} alt={name} />}
        <div className="username">{name}</div>
        {showFollow && (
          <Button
            variant={followBtn}
            size="small"
            sx={{ marginLeft: 'auto' }}
            onClick={handleFollowBtnClick}
          >
            {followed && 'followed'} {!followed && 'follow'}
          </Button>
        )}
      </div>
    </div>
  );
}

export default UserRow;
