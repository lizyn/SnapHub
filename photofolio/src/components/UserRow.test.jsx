/**
 * @jest-environment jsdom
 */

import React from 'react';
// import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import UserRow from './UserRow';

// snapshot testing
test('Post Modal matches snapshot', () => {
  const component = renderer.create(
    <UserRow avatar="" name="" userId={1} showFollow />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders component with Follow button', () => {
  render(<UserRow avatar="" name="" userId={1} showFollow />);
  const followBtn = screen.getByText(/follow/);
  expect(followBtn).toBeInTheDocument();
});

test('without Follow button', () => {
  render(<UserRow avatar="" name="" userId={1} />);
  const followBtn = screen.queryByText(/follow/);
  expect(followBtn).not.toBeInTheDocument();
});

it('test', async () => {
  render(<UserRow avatar="avatar" name="name" userId={1} showFollow />);
  // manually trigger the callback
  const followBtn = screen.getByRole('button', { name: 'follow' });
  userEvent.click(followBtn);
  const unfollowBtn = screen.getByRole('button', { name: 'followed' });
  userEvent.click(unfollowBtn);
});
