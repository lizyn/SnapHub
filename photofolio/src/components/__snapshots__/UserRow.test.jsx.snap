// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Modal matches snapshot 1`] = `
<div
  className="user-row"
>
  <div
    className="container"
  >
    <img
      alt=""
      className="profile-pic"
      src=""
    />
    <div
      className="username"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k62pt1-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
       
      follow
    </button>
  </div>
</div>
`;
